edk2toolext/edk2_logging.py

```diff
@@ -231,6 +231,7 @@ def output_compiler_error(match, line, start_txt="Compiler"):
     build_py_error_exp = re.compile(r"error (\d+)E:")
     linker_error_exp = re.compile(r"error LNK(\d+):")
     warning_exp = re.compile(r"warning [A-Z]?(\d+):")
+    rust_error_exp = re.compile(r"^(-->|\d+\s*\||error:)")
```

Comment:
Could you add a few tests?

Reply:
the regex was reduced to only `error:`, `error[errotype]:` and `-->` to reduce conflicts with catching unexpected c code. Example in the PR summary. 

Tests have been added. Thanks!

Reply:
I have added tests. Thanks!